### PR TITLE
New feature: You can choose smoothing (ie 5 year smoothing window) and set custom plot labels for the legend.

### PR DESCRIPTION
**Pull request by ledm**
Added a new feature today:
- you can now add bespoke custom labels for comparison plots. Previously the legend just used the job ID, now you can put specific text instead. Don't need to check the table at the top now. Much more useful. These values are set in the job definitions in the input_yml file.

- Added the option to make additional plots with smoothing windows. This is set at the key_file level, so you can just add it for some of the analyse as needed. I've added them to the AMOC & Drake passage analyses, but no others.

### CODE REVIEW

File: bgcval2/download_from_mass.py [hunk 1]
```diff
@@ -503,8 +503,14 @@ def download_from_mass(
...
 
-        # check destination permissions:
-        i_can_write_this = os.access(shared_file_path, os.W_OK)
+        # check destination existance and permissions
+        if os.path.exists(shared_file_path):
+            # check destination permissions
+            i_can_write_this = os.access(shared_file_path, os.W_OK)
+        else:
+            # File doesn't exist!
+            i_can_write_this = True
```
Comment (valeriupredoi) [inline bgcval2/download_from_mass.py:512]: well, you can try :grin:

File: bgcval2/timeseries/timeseriesPlots.py [hunk 2]
```diff
@@ -702,9 +702,10 @@ def multitimeseries(
             'lime',
         ],
         plotStyle='Together',
-        lineStyle='',
+        smoothing='',
```
Comment (valeriupredoi) [inline bgcval2/timeseries/timeseriesPlots.py:705]:
```suggestion
        smoothing='',
```

**Review by valeriupredoi — APPROVED** [review]
looking good, but do you not want to add a couple bits about `smoothing` in the docs? Good weekend, bud :beer:

**Comment by ledm** [comment]
> looking good, but do you not want to add a couple bits about `smoothing` in the docs? Good weekend, bud 🍺

Documentation eh? Novel idea. I'll give it a go.

**Comment by valeriupredoi** [comment]
> > looking good, but do you not want to add a couple bits about `smoothing` in the docs? Good weekend, bud 🍺
> 
> Documentation eh? Novel idea. I'll give it a go.

was thinking the README, not anything more controversial and revolutionary as an actual `doc` (tho we do have the sphinx platform working, just need to add content)

File: README.md [hunk 3]
```diff
@@ -500,6 +507,26 @@ basic functions such as multiply by or add to, or `noChange`, which can all be
...
+    - Note that this is a pre-curated list, so you can't use it to select a specific depth (like `327m` or something)
+  - `regions`:
+    - A list of regions to investigate.
+    - This is a pre-curated list, and they are defined in so you can't.
+    - These regions are defined in the file `bgcval2/bgcvaltools/makeMask.py`.
+  -  `smoothings`:
+    - This is the smoothing function (if any) to apply to the data before plotting it.
+    - The smoothing it added before plotting, but after saving the shelve file, so it doesn't impact the data.
+    - No smoothing is `DataOnly`, which is also the default behaviour. 
+    - If several smoothing options are added here, bgcval2 will generate a comparison plot for each one.
```
Comment (ledm) [inline README.md:523]:
```suggestion
    - If several smoothing options are added here, bgcval2 will generate a comparison plot for each one.
    - The smoothings are defined and performed in `bgcval2/timeseries/timeseriesPlots.py`
```

**Comment by ledm** [comment]
Added some documentation. Happy to merge?

**Comment by valeriupredoi** [comment]
yessir! Nice one :beer:

**Pull request merged**